File: app/src/main/java/com/afkanerd/deku/DefaultSMS/ComposeNewMessageActivity.java
package com.afkanerd.deku.DefaultSMS;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;

import com.afkanerd.deku.DefaultSMS.Models.Contacts.Contacts;
import com.afkanerd.deku.DefaultSMS.Models.Contacts.ContactsRecyclerAdapter;
import com.afkanerd.deku.DefaultSMS.ViewModels.ContactsViewModel;
import com.google.android.material.textfield.TextInputEditText;
import com.google.i18n.phonenumbers.NumberParseException;

import java.util.List;

public class ComposeNewMessageActivity extends AppCompatActivity {

    ContactsViewModel contactsViewModel;
    RecyclerView contactsRecyclerView;
    ContactsRecyclerAdapter contactsRecyclerAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_compose_new_message);

        Toolbar toolbar = (Toolbar) findViewById(R.id.compose_new_message_toolbar);
        setSupportActionBar(toolbar);
        // Get a support ActionBar corresponding to this toolbar
        ActionBar ab = getSupportActionBar();
        // Enable the Up button
        ab.setDisplayHomeAsUpEnabled(true);

        ab.setTitle(getString(R.string.search_title));

        contactsViewModel = new ViewModelProvider(this).get(
                ContactsViewModel.class);

        contactsRecyclerAdapter = new ContactsRecyclerAdapter( this);

        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this,
                LinearLayoutManager.VERTICAL, false);

        contactsRecyclerView = findViewById(R.id.compose_new_message_contact_list_recycler_view);
        contactsRecyclerView.setLayoutManager(linearLayoutManager);
        contactsRecyclerView.setAdapter(contactsRecyclerAdapter);

        contactsViewModel.getContacts(getApplicationContext()).observe(this, new Observer<List<Contacts>>() {
            @Override
            public void onChanged(List<Contacts> contacts) {
                contactsRecyclerAdapter.submitList(contacts);
            }
        });

        TextInputEditText textInputEditText = findViewById(R.id.compose_new_message_to);
        textInputEditText.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {

            }

            @Override
            public void afterTextChanged(Editable s) {
                new Thread(new Runnable() {
                    @Override
                    public void run() {
                        if (s != null && !s.toString().isEmpty()) {
                            try {
                                contactsViewModel.filterContact(s.toString());
                            } catch (NumberParseException e) {
                                e.printStackTrace();
                            }
                        }
                        else {
                            try {
                                contactsViewModel.filterContact("");
                            } catch (NumberParseException e) {
                                e.printStackTrace();
                            }
                        }
                    }
                }).start();
            }
        });

        _checkSharedContent();
    }

    private void _checkSharedContent() {
        if (Intent.ACTION_SEND.equals(getIntent().getAction()) && getIntent().getType() != null) {
            if ("text/plain".equals(getIntent().getType())) {
                String sharedSMS = getIntent().getStringExtra(Intent.EXTRA_TEXT);
                contactsRecyclerAdapter.setSharedMessage(sharedSMS);
            }
        }
    }
}

File: app/src/main/java/com/afkanerd/deku/DefaultSMS/ComposeNewMessageActivity.kt
package com.afkanerd.deku.DefaultSMS

import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.View
import android.widget.Toolbar
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.afkanerd.deku.DefaultSMS.AdaptersViewModels.ContactsRecyclerAdapter
import com.afkanerd.deku.DefaultSMS.AdaptersViewModels.ContactsViewModel
import com.afkanerd.deku.Modules.ThreadingPoolExecutor
import com.google.android.material.textfield.TextInputEditText
import com.google.i18n.phonenumbers.NumberParseException


class ComposeNewMessageActivity : CustomAppCompactActivity() {
    private val contactsViewModel: ContactsViewModel by viewModels()
    private lateinit var contactsRecyclerView: RecyclerView
    private val contactsRecyclerAdapter: ContactsRecyclerAdapter = ContactsRecyclerAdapter()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_compose_new_message)

        val myToolbar: androidx.appcompat.widget.Toolbar = findViewById(R.id.compose_new_message_toolbar)
        setSupportActionBar(myToolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.title = getString(R.string.compose_new_message_title)

        val linearLayoutManager = LinearLayoutManager( this,
            LinearLayoutManager.VERTICAL, false )

        contactsRecyclerView = findViewById(R.id.compose_new_message_contact_list_recycler_view)
        contactsRecyclerView.setLayoutManager(linearLayoutManager)
        contactsRecyclerView.setAdapter(contactsRecyclerAdapter)

        contactsViewModel.getContacts(applicationContext).observe(this, Observer {
            contactsRecyclerAdapter.submitList(it)
        })

        val textInputEditText = findViewById<TextInputEditText>(R.id.compose_new_message_to)
        textInputEditText.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
            }

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
            }

            override fun afterTextChanged(s: Editable) {
                ThreadingPoolExecutor.executorService.execute {
                    try {
                        if(s.isNotEmpty())
                            contactsViewModel.filterContact(applicationContext, s.toString())
                        else contactsViewModel.filterContact(applicationContext, "")
                    } catch (e: NumberParseException) {
                        e.printStackTrace()
                    }
                }
            }
        })

        checkSharedContent()
    }

    private fun checkSharedContent() {
        if (Intent.ACTION_SEND == intent.action && intent.type != null) {
            if ("text/plain" == intent.type) {
                val sharedSMS = intent.getStringExtra(Intent.EXTRA_TEXT)
                contactsRecyclerAdapter!!.setSharedMessage(sharedSMS)
            }
        }
    }

    override fun onResume() {
        super.onResume()
        findViewById<View>(R.id.compose_new_message_to).requestFocus()
    }
}